lib/rubycritic/source_control_systems/git.rb

```diff
@@ -52,11 +52,15 @@ def travel_to_head
       end
 
       def self.switch_branch(branch)
-        uncommitted_changes? ? abort('Uncommitted changes are present.') : `git checkout #{branch}`
+        uncommitted_changes? ? abort("Uncommitted changes are present: #{uncommitted_changes}") : `git checkout #{branch}`
```

Comment:
This can be changed into the following to avoid big lines:

```ruby
dirty = !uncommitted_changes.blank?
abort("Uncommitted changes are present: #{uncommitted_changes}") if dirty? 

`git checkout #{branch}`
```

Reply:
`self.uncommitted_changes?` can then be deleted, or change `self.uncommitted_changes` so that it memoizes the result (because calling git via an OS call is expensive): 

```ruby
def self.uncommitted_changes
    return @uncommitted_changes if defined? @uncommitted_changes
    @uncommitted_changes = git('diff-index HEAD --').chomp! || ''
 end
```

(from the top of my head I don't think it will be a problem to have this value cached 🤔 )